### PR TITLE
added list_data_ids() convenience function

### LINKED ISSUE #776: Add `DataStore.list_data_ids()`
**Is your feature request related to a problem? Please describe.**

`DataStore.get_data_ids()` returns an iterator for a good reason. A store might comprise thousands of datasets!

However, most of our stores comprise tens or hundreds of entries. And in a Jupyter Notebook we always have to type

```python
list(store.get_data_ids())
```

It would be more convenient and comprehensive if we could just write

```python
store.list_data_ids()
```

**Describe the solution you'd like**

- [ ]  Add `list_data_ids()` with default implementation `return list(iter_data_ids())`

In a follow-up of this issue, we could make things more clear by

- [ ]  Deprecate `get_data_ids()`
- [ ]  Add `iter_data_ids()` with default implementation `return get_data_ids()`
- [ ]  Replace all other usages of `get_data_ids()` by `iter_data_ids()`

### PR DESCRIPTION
**Pull request by forman**
Closes #776

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/core/store/store.py [hunk 1]
```diff
@@ -237,6 +237,29 @@ def get_data_ids(self,
...
+        than an iterator.
+
+        :param data_type: If given, only data identifiers that are
+            available as this type are returned.
+            If this is omitted, all available data identifiers are returned.
+        :param include_attrs: A sequence of names of attributes to
+            be returned for each dataset identifier.
+            If given, the store will attempt to provide the set of
+            requested dataset attributes in addition to the data ids.
+            (added in xcube 0.8.0)
```
Comment (TonioF) [inline xcube/core/store/store.py:255]:
```suggestion
            (added in xcube 0.8.0)
```

File: test/core/store/fs/test_registry.py [hunk 2]
```diff
@@ -381,6 +381,7 @@ def _assert_dataset_supported(
             data_store.get_data_types_for_data(data_id)
         self.assertEqual(False, data_store.has_data(data_id))
         self.assertNotIn(data_id, set(data_store.get_data_ids()))
+        self.assertNotIn(data_id, set(data_store.list_data_ids()))
```
Comment (TonioF) [inline test/core/store/fs/test_registry.py:384]:
```suggestion
        self.assertNotIn(data_id, data_store.list_data_ids())
```

File: test/core/store/fs/test_registry.py [hunk 3]
```diff
@@ -426,6 +428,7 @@ def _assert_dataset_supported(
...
```
Comment (TonioF) [inline test/core/store/fs/test_registry.py:431]:
```suggestion
        self.assertNotIn(data_id, data_store.list_data_ids())
```

**Review by TonioF — APPROVED** [review]
Only minor comments.

**Comment by forman** [comment]
@TonioF Merci!

**Pull request merged**